compiler/renderer.nim

```diff
@@ -103,7 +104,8 @@ proc initSrcGen(g: var TSrcGen, renderFlags: TRenderFlags; config: ConfigRef) =
   g.pendingNL = -1
   g.pendingWhitespace = -1
   g.inGenericParams = false
-  g.config = config
+  g.config = config  
+  g.cache = newIdentCache()
```

Comment:
This can cause subtle bugs, better pass it to ``initSrcGen`` somehow, if really required (I don't think it is).

Reply:
Ideal fix: removed IdentCache from the renderer all together